contracts/proxy/src/contract.rs

```diff
@@ -335,76 +336,111 @@ pub fn execute_update_guardians(
     deps: DepsMut,
     env: Env,
     info: MessageInfo,
-    guardians: Guardians,
-    new_multisig_code_id: Option<u64>,
 ) -> Result<Response, ContractError> {
-    // ensure this is either a direct message from the user
-    // or
-    // ensure this is relayed by a relayer from this proxy
-    let is_user = ensure_is_user(deps.as_ref(), info.sender.as_ref());
-    let is_contract = ensure_is_contract_self(&env, &info.sender);
-    if is_user.is_err() && is_contract.is_err() {
-        is_user?;
-        is_contract?;
-    };
-
-    guardians.verify_guardians(&deps.api.addr_humanize(&USER.load(deps.storage)?.addr)?)?;
+    ensure_is_relayer_or_user(deps.as_ref(), &env, &info.sender)?;
 
     // make sure guardians have not frozen the contract
     if is_frozen(deps.as_ref())? {
         return Err(ContractError::Frozen {});
     }
 
-    // Replace the entire locally stored guardians list
-    let guardians_to_remove = load_canonical_addresses(&deps.as_ref(), GUARDIANS)?;
-    for guardian in guardians_to_remove {
-        GUARDIANS.remove(deps.storage, &guardian);
-    }
-    for guardian in &guardians.addresses {
-        GUARDIANS.save(deps.storage, &deps.api.addr_canonicalize(guardian)?, &())?;
-    }
+    match PENDING_GUARDIAN_ROTATION.may_load(deps.storage)? {
```

Comment:
```suggestion
    let request = PENDING_GUARDIAN_ROTATION.may_load(deps.storage)?.ok_or_else(|| ContractError::GuardianRequestNotFound{})?;
```

This might be easier to read.

Reply:
it looks much better thank for the suggestion!